kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/languages/LanguageImpl.kt

```diff
@@ -23,6 +23,9 @@ class LanguageImpl(
             if (stage == Stage.CREATED) {
                 cachedActivity = activity
             }
+            if ((stage == Stage.DESTROYED) and (cachedActivity == activity)) {
```

Comment:
AFAIK `onDestroy` method call is not guaranteed. What about  `Stage.DESTROYED`?  
May be weak reference on `cachedActivity` is more suitable here?

Reply:
Yes, is better